pkg/resourcecollector/resourcecollector.go

```diff
@@ -187,12 +205,24 @@ func (r *ResourceCollector) objectToBeCollected(
 	object runtime.Unstructured,
 	crbs *rbacv1.ClusterRoleBindingList,
 	namespace string,
+	destCluster bool,
 ) (bool, error) {
 	metadata, err := meta.Accessor(object)
 	if err != nil {
 		return false, err
 	}
 
+	// check migration resources
+	if destCluster {
+		if val, ok := metadata.GetAnnotations()[storkMigrationAnnotation]; ok {
```

Comment:
I think the `ResourceCollector` package shouldn't know about this, it should just return all the objects and the migration controller can decide which ones to filter out.